src/plugins/interactive_view_control/InteractiveViewControl.cc

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <string>
+
+#include <ignition/common/MouseEvent.hh>
+
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/GuiEvents.hh>
+#include <ignition/gui/MainWindow.hh>
+
+#include <ignition/plugin/Register.hh>
+
+#include <ignition/rendering/Camera.hh>
+#include <ignition/rendering/OrbitViewController.hh>
+#include <ignition/rendering/RenderingIface.hh>
+#include <ignition/rendering/RayQuery.hh>
+
+#include "InteractiveViewControl.hh"
+
+/// \brief Private data class for InteractiveViewControl
+class ignition::gui::plugins::InteractiveViewControlPrivate
+{
+  /// \brief
+  public: void OnRender();
+
+  math::Vector3d ScreenToScene(
+      const math::Vector2i &_screenPos) const;
+
+  /// \brief Flag to indicate if mouse event is dirty
+  public: bool mouseDirty = false;
+
+  public: bool blockOrbit = false;
+
+  /// \brief Mouse event
+  public: common::MouseEvent mouseEvent;
+
+  /// \brief Mouse move distance since last event.
+  public: math::Vector2d drag;
+
+  /// \brief User camera
+  public: rendering::CameraPtr camera{nullptr};
+
+  /// \brief View control focus target
+  public: math::Vector3d target;
+
+  /// \brief Camera orbit controller
+  public: rendering::OrbitViewController viewControl;
```

Comment:
Note that we'll also need to support ortho control, added in https://github.com/ignitionrobotics/ign-gazebo/pull/815.

That's where I think that this plugin starts overlapping with `ViewAngle`. I'm not sure if we should put the ortho logic here, but keep the button on `ViewAngle`. I guess we can start with that and see if it becomes too confusing for users?

Reply:
Added ortho control https://github.com/ignitionrobotics/ign-gui/pull/231/commits/c81302dd505d90d69fc872fc722b8de322ccd497